### ISSUE TITLE
Get Full Crud APIs

### ISSUE THREAD

**Issue by ibnshayed** [tiangolo/fastapi#2837]
Is it possible to make something like class or method or function and get the full CRUD APIs of it?

from fastapi import FastAPI

app = FastAPI()

@app.crud("/user"):
class or method:
  user attibutes

only the above things give us the following ( 6 API or more common)

@app.getAll("/user") # get all users api
@app.getById("/user/{id}") # get by something
@app.create("/user")
@app.delete("/user/{id}") # delete by something
@app.update("/user/{id}") # update by something
@app.patch("user/{id}") # partially update by something

**Comment by ycd** [comment]
No, but as far as i have seen, you are trying to create boilerplate code for user management, does [fastapi-users](https://github.com/frankie567/fastapi-users) helps?

**Comment by ibnshayed** [comment]
> No, but as far as I have seen, you are trying to create boilerplate code for user management, does [fastapi-users](https://github.com/frankie567/fastapi-users) helps?

No, I don't say about the user, I would like to say that  Is it possible to make something like class or method or function and get the full CRUD APIs of it? like any other. It can any class or method, 
we just creat 
@app.crud("/book") and get full crud api

**Comment by krishnardt** [comment]
Do you mean entire logical code of an api?

**Comment by insomnes** [comment]
@ibnshayed maybe fastapi-utils can help with class based views:
https://fastapi-utils.davidmontague.xyz/user-guide/class-based-views/

But it is also not totaly like DRF view or something like that.

**Comment by ycd** [comment]
>  I would like to say that Is it possible to make something like class or method or function and get the full CRUD APIs

Interesting idea and i don't think it is a worthwhile feature when you think about how hard to implement something like that, could be a great external package tho. Are there any frameworks that can do that?

**Comment by ibnshayed** [comment]
@ycd Please see Django Rest Framework Viewset https://www.django-rest-framework.org/api-guide/viewsets/

**Comment by ycd** [comment]
Yes, but DRF built on top of Django, it has built in ORM, class-based views, templating etc. Those all features are out of scope for a microframework imo.

**Comment by Kludex** [comment]
This?

```python
from fastapi import FastAPI

app = FastAPI()


@app.api_route("/potato", methods=["GET", "POST", "PUT", "DELETE"])
def home():
    return "Hello World!"
```

**Comment by ycd** [comment]
@ibnshayed i come across this library today. 

See: https://github.com/awtkns/fastapi-crudrouter

**Comment by ibnshayed** [comment]
@ycd joss boss :)

**Comment by DomWeldon** [comment]
I needed to do something similar a while ago that I've been looking to open source. *It needed to support composite primary and foreign keys, and non-PK unique index checks (as in Postgres), which I know a lot of auto-crud apps don't do (and the DRF is awful at).* 

The code is below, I'll try and create a repo for it later today, the tests are currently coupled to my use case.

https://gist.github.com/DomWeldon/adb17318202dd0b673a0753e37c46b4b

If anyone is interested in picking this up / helping out on this, I'd be very happy to collaborate.
